### DISCUSSION TITLE
The accuracy of loading DSM terrain data is missing[Bug]

### DISCUSSION THREAD

**Discussion by Paweled** [visgl/deck.gl#6582] [Q&A]
### Description

I use import {terrainlayer} from '@ deck gl/geo-layers'; Render terrain data.
Setting: meshmaxerror = 0；
The renderings are as follows:
![image](https://user-images.githubusercontent.com/45220025/150111201-a81a4023-63bb-4dc1-8971-0938757202b8.png)

It looks like a square, the effect is not very good-looking, and it feels like the lack of precision.

The coordinate format of this is UTM,
coordinateSystem= COORDINATE_ SYSTEM. CARTESIAN,

### Flavors

- [X] React
- [ ] Python/Jupyter notebook
- [ ] MapboxLayer
- [x] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] DeckLayer/DeckRenderer for ArcGIS

### Expected Behavior

Each mesh should be triangular。
The effect should be like this:
![image](https://user-images.githubusercontent.com/45220025/150113494-d7b0ed0e-a21d-42f7-84f1-44b2573ac684.png)

### Steps to Reproduce

Setting: meshmaxerror = 0；

### Environment

- Framework version: deck.gl@8.7.0-alpha.11
- Browser: Chrome 97.0.4692.71
- OS:

**Comment by ibgreen** [comment]
@kylebarron @igorDykhta @chrisgervang

**Comment by kylebarron** [comment]
`meshMaxError` should never be 0 if you want a visually-appealing terrain mesh. When set to `0` the triangles can't approximate the underlying gridded terrain data, and you actually end up creating triangles that exactly reproduce the underlying square pixels.

**Comment by Paweled** [comment]
The above two figures set meshmaxerror = 0;
But the second effect looks better.

The first one uses UTM coordinate format,
![image](https://user-images.githubusercontent.com/45220025/150262116-bf4adb56-672f-4373-be8c-218c2363b12d.png)
![image](https://user-images.githubusercontent.com/45220025/150262149-4df88de8-5100-46f5-85a4-b4ac797397de.png)


The second is the result of my modification of the source code. I only convert the UTM coordinates to the distance (m), and then make the tile offset (WGS84 coordinate format)
![image](https://user-images.githubusercontent.com/45220025/150262223-9701a004-43e2-403d-91f9-2f5961dd88af.png)
![image](https://user-images.githubusercontent.com/45220025/150262299-578e5c81-6df1-49b2-a417-d87973b662d3.png)
@kylebarron

**Comment by kylebarron** [comment]
I'm still not sure I entirely follow. Are you saying there's an error with how the mesh is generated or with how it's rendered (or both?).

**Comment by Paweled** [comment]
> I'm still not sure I entirely follow. Are you saying there's an error with how the mesh is generated or with how it's rendered (or both?).

The triangular meshes（indices） of these two methods are the same, and the only difference is the coordinates and rendering method（coordinateSystem）.

I don't know how to calculate this UTM coordinate. Can it be converted to WGS84 coordinate? How should I convert?

**Comment by Pessimistress** [comment]
I'm moving this thread to Discussions. As @kylebarron already pointed out, you should set `meshMaxError` to a number larger than 0. If you still believe there is a bug in deck.gl, please create a Codepen/Code Sandboxthat reproduce your issue.
